app/src/main/java/ch/epfl/sdp/Map/PathsHandler.java

```diff
@@ -0,0 +1,136 @@
+package ch.epfl.sdp.Map;
+
+import android.graphics.Color;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.GeoPoint;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.mapbox.geojson.Point;
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.FeatureCollection;
+import com.mapbox.geojson.LineString;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.LineLayer;
+import com.mapbox.mapboxsdk.style.layers.Property;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import ch.epfl.sdp.Callback;
+import ch.epfl.sdp.R;
+import ch.epfl.sdp.contamination.ConcreteDataReceiver;
+import ch.epfl.sdp.contamination.GridFirestoreInteractor;
+
+import static com.mapbox.mapboxsdk.style.layers.Property.NONE;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.visibility;
+
+/**
+ * This class is used to display the user's last positions as a line on the map,
+ * as well as points of met infected users.
+ */
+public class PathsHandler extends Fragment {
+    private MapboxMap map;
+    public List<Point> pathCoordinates; // public for testing
+    private FirebaseFirestore db = FirebaseFirestore.getInstance(); // we don't use ConcreteFirestoreInteractor because we want to do more specific op
+    private MapFragment parentClass;
+    public double latitude; //public fir testing
+    public double longitude; // public for testing
+
+    // default access restriction for now, could be package-private, depending on how we finally decide to organize files
+    public static final String PATH_LAYER_ID = "linelayer"; // public for testing
+    static final String PATH_SOURCE_ID = "line-source";
+
+    public PathsHandler(@NonNull MapFragment parentClass, @NonNull MapboxMap map) { // public for testing
+        this.parentClass = parentClass;
+        this.map = map;
+        initFirestorePathRetrieval(this::getPathCoordinates);
+    }
+
+    // public for now, could be package-private, depending on how we finally decide to organize files
+    public void setCameraPosition() {
+        CameraPosition position = new CameraPosition.Builder()
+                .target(new LatLng(latitude, longitude))
+                .zoom(7)
```

Comment:
Magic constant, for greater maintainability it's better to use a final private variable